modules/plugins/formatter/conform-nvim/conform-nvim.nix

```
@@ -30,27 +27,68 @@ in {
         description = "Default values when calling `conform.format()`";
       };
 
-      format_on_save = mkOption {
-        type = nullOr attrs;
-        default = {
+      format_on_save = let
+        defaultFormatOnSaveOpts = {
           lsp_format = "fallback";
           timeout_ms = 500;
         };
-        description = ''
-          Table that will be passed to `conform.format()`. If this
-          is set, Conform will run the formatter on save.
-        '';
-      };
+      in
+        mkOption {
+          type = nullOr (either attrs luaInline);
+          default =
+            mkLuaInline
+            # lua
+            ''
+              function()
+                if not vim.g.formatsave or vim.b.disableFormatSave then
+                  return
+                else
+                  return ${toLuaObject defaultFormatOnSaveOpts}
+                end
+              end
+            '';
+          description = ''
+            Table or function(lualinline) that will be passed to `conform.format()`. If this
```

```suggestion
            Table or function(lualinline) that will be passed to `conform.format()`. If this
```

Is this a function that takes `luainline` as an argument?

---

>  Is this a function that takes luainline as an argument?

it can take both

a table or a function that returns table

---

Okay, I'm mostly confused about the role of the paranthesis in the description. Does it indicate function arguments? 

I think it's better to clarify the wording to state the types properly.


```suggestion
            Attribute set or or a Lua function that will be passed to `conform.format()`.
```

Or similar.